File: src/lib/auth-form.ts
export function looksLikeEmail(value: string) {
  return /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(value.trim());
}

export function emailFieldError(value: string) {
  const trimmed = value.trim();
  if (!trimmed) return "Enter your email.";
  if (!looksLikeEmail(trimmed)) return "Enter a valid email address.";
  return null;
}

export function passwordFieldError(value: string, { creating }: { creating: boolean }) {
  if (!value) return creating ? "Choose a password." : "Enter your password.";
  if (creating && value.length < 8) return "Use at least 8 characters.";
  return null;
}

export function confirmFieldError(password: string, confirm: string) {
  if (!confirm) return "Confirm your password.";
  if (password !== confirm) return "Those passwords do not match.";
  return null;
}

File: android/core/src/main/kotlin/com/eventview/core/AuthForm.kt
package com.eventview.core

/** Mirrors `src/lib/auth-form.ts`. */
object AuthForm {
    fun looksLikeEmail(value: String): Boolean {
        return Regex("^[^\\s@]+@[^\\s@]+\\.[^\\s@]+$").matches(value.trim())
    }

    fun emailFieldError(value: String): String? {
        val trimmed = value.trim()
        if (trimmed.isEmpty()) return "Enter your email."
        if (!looksLikeEmail(trimmed)) return "Enter a valid email address."
        return null
    }

    fun passwordFieldError(value: String, creating: Boolean): String? {
        if (value.isEmpty()) return if (creating) "Choose a password." else "Enter your password."
        if (creating && value.length < 8) return "Use at least 8 characters."
        return null
    }

    fun confirmFieldError(password: String, confirm: String): String? {
        if (confirm.isEmpty()) return "Confirm your password."
        if (password != confirm) return "Those passwords do not match."
        return null
    }

    fun displayName(name: String, email: String): String {
        val trimmed = name.trim()
        if (trimmed.isNotEmpty()) return trimmed.take(80)
        val local = email.trim().substringBefore("@")
        return local.ifEmpty { "Host" }
    }
}
